v2/client/src/components/common/Feedback/index.js

```diff
@@ -0,0 +1,180 @@
+// renders chart js graphs for trainer feedback
+
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { HorizontalBar } from 'react-chartjs-2';
+import Spin from '../Spin';
+
+import {
+  Wrapper,
+  ChartWrapper,
+  Description,
+  Container,
+  HeadlineDiv,
+} from './Feedback.style';
+
+import { fetchTrainerFeedback as fetchTrainerFeedbackAction } from '../../../actions/users';
+
+class TrainerFeedbackOverall extends Component {
+  componentDidMount() {
+    const { fetchTrainerFeedback, trainerId, sessionId } = this.props;
+    fetchTrainerFeedback(trainerId, sessionId);
+  }
+
+  sum = arr => arr.reduce((a, b) => a + b);
+
+  render() {
+    const { feedbackData, loaded } = this.props;
+    const chartsData = [];
+    const items = Object.keys(feedbackData);
+    // loop over question groups
+    items.map(text => {
```

Comment:
triple nested loop is like very bad, O(n^3) performance

see if you can do changes to data structure or separate out some of them out